tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -270,6 +270,17 @@ public TracerSettings(IConfigurationSource source)
             {
                 HttpClientExcludedUrlSubstrings = TrimSplitString(urlSubstringSkips.ToUpperInvariant(), commaSeparator);
             }
+
+            var dbmPropagationMode = source?.GetString(ConfigurationKeys.DbmPropagationMode);
+            if (int.TryParse(dbmPropagationMode, out var parsedInt))
```

Comment:
This is a smart way of preventing users from using integers since they are valid in `Enum.TryParse`, but there's another weird `Enum.TryParse` behavior that could still result in an invalid value:
```
DD_DBM_PROPAGATION_MODE=Service, Full
```

This will split the string and parse the parts separately, then combine them using the bitwise OR operator, even if `DbmPropagationLevel` is not marked with `[Flags]`. The result will be the same as
```
DbmPropagationMode = DbmPropagationLevel.Service | DbmPropagationLevel.Full;
// or
DbmPropagationMode = (DbmPropagationLevel)(1 | 2);
// or
DbmPropagationMode = (DbmPropagationLevel)3;
```